### PR TITLE
Use ActiveSupport::Notifications to broadcast request details

### PR DESCRIPTION
**Pull request by rhearnorth**
Add functionality to broadcasting requests details

### CODE REVIEW

File: lib/magento/connection.rb [hunk 1]
```diff
@@ -74,5 +77,19 @@ def log_call(message)
...
+
+
+      def notify_about_request(method, response, *args)
+        ActiveSupport::Notifications.instrument("request.magento_api") do |payload|
+          payload[:method]        = method
+          payload[:request_uri]   = config[:uri].to_s
+          payload[:request_body]  = if args.is_a?(Array) &&
+                                       args[0].is_a?(Array) &&
+                                       args[0][1].is_a?(Hash)
+                                      args[0][1].to_json
```
Comment (gherry) [inline lib/magento/connection.rb]: This just seems to specific. What is the drawback of just showing it always in JSON regardless of the args?

Comment (rhearnorth) [inline lib/magento/connection.rb]: `MagentoAPI::Product.find(product_id)` args is `[[product_id]]`
`MagentoAPI::Product.update(product_id, sku: 'test-sku' )` args is `[[product_id, {sku: 'test-sku'}]]`

File: lib/magento/connection.rb [hunk 2]
```diff
@@ -74,5 +77,19 @@ def log_call(message)
...
+                                    end
+          payload[:response_body] = response.to_s
```
Comment (bradleypriest) [inline lib/magento/connection.rb]: Do we want this (I think it's a hash) or do we want the raw response from `MagentoAPI::Base.connection.client.last_response`?

Comment (bradleypriest) [inline lib/magento/connection.rb]: Personally, I say hash for now, we can always go for extra fidelity later if we need it (it'd be a lot more data)

Comment (rhearnorth) [inline lib/magento/connection.rb]: this is sample data 
```ruby
{:method=>"product.info", :request_uri=>"http://52.74.60.126/index.php/api/xmlrpc", :request_body=>nil, :response_body=>"{\"product_id\"=>\"234\", \"sku\"=>\"test-test\", \"set\"=>\"13\", \"type\"=>\"simple\", \"categories\"=>[\"15\"], \"websites\"=>[\"1\"], \"type_id\"=>\"simple\", \"name\"=>\"Slim fit Dobby Oxford Shirt\", \"description\"=>\"Tailored/Slim fit. Long sleeves. Button cuff. Cotton. Imported.\", \"short_description\"=>\"A bold hue and understated dobby detail bring refined nuance to this modern dress shirt. \", \"weight\"=>\"1.0000\", \"news_from_date\"=>nil, \"old_id\"=>nil, \"news_to_date\"=>nil, \"status\"=>\"1\", \"url_key\"=>\"slim-fit-dobby-oxford-shirt\", \"visibility\"=>\"1\", \"country_of_manufacture\"=>nil, \"url_path\"=>\"slim-fit-dobby-oxford-shirt.html\", \"category_ids\"=>[\"15\"], \"required_options\"=>\"0\", \"has_options\"=>\"0\", \"image_label\"=>nil, \"small_image_label\"=>nil, \"thumbnail_label\"=>nil, \"created_at\"=>\"2013-05-03T01:48:13-04:00\", \"updated_at\"=>\"2016-10-19 03:24:48\", \"price\"=>\"175.0000\", \"group_price\"=>[], \"special_price\"=>nil, \"special_from_date\"=>nil, \"minimal_price\"=>nil, \"special_to_date\"=>nil, \"tier_price\"=>[], \"msrp_enabled\"=>\"2\", \"msrp_display_actual_price_type\"=>\"4\", \"msrp\"=>nil, \"tax_class_id\"=>\"2\", \"meta_title\"=>nil, \"meta_keyword\"=>nil, \"meta_description\"=>nil, \"is_recurring\"=>nil, \"recurring_profile\"=>nil, \"custom_design\"=>nil, \"custom_design_from\"=>nil, \"custom_design_to\"=>nil, \"custom_layout_update\"=>nil, \"page_layout\"=>\"one_column\", \"options_container\"=>\"container1\", \"gift_message_available\"=>nil, \"gift_wrapping_available\"=>nil, \"gift_wrapping_price\"=>nil, \"color\"=>\"27\", \"occasion\"=>\"29\", \"apparel_type\"=>\"41\", \"sleeve_length\"=>\"47\", \"fit\"=>\"50\", \"size\"=>\"80\", \"length\"=>nil, \"gender\"=>\"93\", \"sizes\"=>nil}"}
```

Comment (rhearnorth) [inline lib/magento/connection.rb]: we keep `response_body` in kibana as string (for unexpected error as well)

Comment (bradleypriest) [inline lib/magento/connection.rb]: So you do want to go with the last_response object then?
It feels weird to store a ruby hash as it's stringified form, it's not like that's what the API response is

**Pull request merged**